### PR TITLE
[solaredge] simplified internal channel modelling

### PR DESCRIPTION
**Pull request by alexf2015**
removed duplicated modelling of channels in JAVA code. This should be easier to maintain.
Also improved code quality by fixing a lot of compiler warnings (most related to missing @NonNullByDefault annotations).

There is no functional change implemented in this PR.

### CODE REVIEW

File: bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/AggregateDataResponsePublicApi.java [hunk 1]
```diff
@@ -12,346 +12,45 @@
...
-
-    private static final String METER_TYPE_PRODUCTION = "Production";
-    private static final String METER_TYPE_CONSUMPTION = "Consumption";
-    private static final String METER_TYPE_SELFCONSUMPTION = "SelfConsumption";
-    private static final String METER_TYPE_IMPORT = "Purchased";
-    private static final String METER_TYPE_EXPORT = "FeedIn";
+@NonNullByDefault
+public class AggregateDataResponsePublicApi {
 
+    @NonNullByDefault
```
Comment (fwolter) [inline bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/AggregateDataResponsePublicApi.java:28]: Null annotations on inner classes are not necessary. There is a bug in checkstyle, generating a false positive warning, which can safely be ignored. Same for the others.

File: bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/LiveDataResponseTransformer.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    public Map<Channel, State> transform(LiveDataResponseMeterless response) {
+        Map<Channel, State> result = new HashMap<>(20);
+        Overview overview = response.getOverview();
+
+        if (overview != null) {
+            if (overview.currentPower != null) {
+                putPowerType(result, channelProvider.getChannel(CHANNEL_GROUP_LIVE, CHANNEL_ID_PRODUCTION),
+                        overview.currentPower.power, UNIT_W);
```
Comment (fwolter) [inline bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/LiveDataResponseTransformer.java:50]: These massive compiler warnings are not nice. You could store it to a local variable, which would be one extra line each.
```suggestion
            Power localCurrentPower = overview.currentPower;
            if (localCurrentPower != null) {
                putPowerType(result, channelProvider.getChannel(CHANNEL_GROUP_LIVE, CHANNEL_ID_PRODUCTION),
                        localCurrentPower.power, UNIT_W);
```

File: bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/callback/AbstractCommandCallback.java [hunk 3]
```diff
@@ -172,13 +190,22 @@ public CommunicationStatus getCommunicationStatus() {
...
+     * null checks which are not unnecessary
+     *
+     * @param <T>
+     * @param json
+     * @param classOfT
+     * @return
+     * @throws JsonSyntaxException
+     */
+    protected <T> @Nullable T fromJson(String json, Class<T> classOfT) throws JsonSyntaxException {
+        return gson.fromJson(json, (Type) classOfT);
```
[the remark this answers is not in the document]
Comment (alexf2015) [reply bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/callback/AbstractCommandCallback.java:209]: you are right, this does not make any sense. It is a c&p mistake.

File: bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/AggregateDataResponsePublicApi.java (hunk 1 above)
Comment (alexf2015) [reply bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/AggregateDataResponsePublicApi.java:28]: I recognized that there are some differences between warnings shown by Eclipse and those shown by the mvn build process.
But I thought that Eclipse was wrong. I will remove these.

File: bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/LiveDataResponseTransformer.java (hunk 2 above)
Comment (alexf2015) [reply bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/LiveDataResponseTransformer.java:50]: Hmm it seems that I have overlooked this. In the other transformer class I solved it exactly like you suggested.

**Pull request merged**